BuildScript/BuildTargets/Testing.targets

```diff
@@ -127,16 +119,31 @@
       <_mergedTestCategoriesToExclude>$(_mergedTestCategoriesToExclude.Trim(','))</_mergedTestCategoriesToExclude>
 
-      <_excludeCategoriesCallArgumentString></_excludeCategoriesCallArgumentString>
-      <_excludeCategoriesCallArgumentString Condition="'$(_mergedTestCategoriesToExclude)' != ''">/exclude=$(_mergedTestCategoriesToExclude)</_excludeCategoriesCallArgumentString>
+      <_mergedTestCategoriesToInclude>$(TestCategoriesToInclude.Replace(';', ',')),$(_includeCategoriesInTestConfiguration)</_mergedTestCategoriesToInclude>
+      <_mergedTestCategoriesToInclude>$(_mergedTestCategoriesToInclude.Trim(','))</_mergedTestCategoriesToInclude>
     </PropertyGroup>
 
+    <ItemGroup>
+      <SplitCategoriesToExclude Include="$(_mergedTestCategoriesToExclude.Split(','))"/>
+      <SplitCategoriesToInclude Include="$(_mergedTestCategoriesToInclude.Split(','))"/>
+      <ExludeFilters Include="cat != %(SplitCategoriesToExclude.Identity) "/>
+      <InludeFilters Include="cat == %(SplitCategoriesToInclude.Identity) "/>
+    </ItemGroup>
+
     <PropertyGroup>
-      <_mergedTestCategoriesToInclude>$(TestCategoriesToInclude.Replace(';', ',')),$(_includeCategoriesInTestConfiguration)</_mergedTestCategoriesToInclude>
-      <_mergedTestCategoriesToInclude>$(_mergedTestCategoriesToInclude.Trim(','))</_mergedTestCategoriesToInclude>
+      <_excludeCategoriesCallArgumentString>@(ExludeFilters)</_excludeCategoriesCallArgumentString>
+      <_excludeCategoriesCallArgumentString>$(_excludeCategoriesCallArgumentString.Replace(";", " &amp;&amp; "))</_excludeCategoriesCallArgumentString>
+
+      <_includeCategoriesCallArgumentString>@(InludeFilters)</_includeCategoriesCallArgumentString>
+      <_includeCategoriesCallArgumentString>$(_includeCategoriesCallArgumentString.Replace(";", " &amp;&amp; "))</_includeCategoriesCallArgumentString>
+    </PropertyGroup>
 
-      <_includeCategoriesCallArgumentString></_includeCategoriesCallArgumentString>
-      <_includeCategoriesCallArgumentString Condition="'$(_mergedTestCategoriesToInclude)' != ''">/include=$(_mergedTestCategoriesToInclude)</_includeCategoriesCallArgumentString>
+    <PropertyGroup>
+      <_nunitTestFilter></_nunitTestFilter>
+      <_nunitTestFilter Condition="$(_mergedTestCategoriesToExclude)!='' And $(_mergedTestCategoriesToInclude)!=''">$(_excludeCategoriesCallArgumentString) &amp;&amp; $(_includeCategoriesCallArgumentString)</_nunitTestFilter>
+      <_nunitTestFilter Condition="$(_mergedTestCategoriesToExclude)!='' And $(_mergedTestCategoriesToInclude)==''">$(_excludeCategoriesCallArgumentString)</_nunitTestFilter>
+      <_nunitTestFilter Condition="$(_mergedTestCategoriesToExclude)=='' And $(_mergedTestCategoriesToInclude)!=''">$(_includeCategoriesCallArgumentString)</_nunitTestFilter>
+      <_nunitTestFilter Condition="$(_nunitTestFilter)!=''">--where "$(_nunitTestFilter)"</_nunitTestFilter>
```

Comment:
should we add a comment sample that shows the final output for include+exclude with multiple categorires? It's a bit mroe involved than before.